transport/src/androidUnitTest/kotlin/transport/core/messagingclient/MCCardsAndCarouselTests.kt

```diff
@@ -0,0 +1,65 @@
+package transport.core.messagingclient
+
+import com.genesys.cloud.messenger.transport.core.Message
+import com.genesys.cloud.messenger.transport.core.MessageStore
+import com.genesys.cloud.messenger.transport.shyrka.send.StructuredMessage
+import com.genesys.cloud.messenger.transport.utility.QuickReplyTestValues
+import io.mockk.every
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import org.junit.Test
+import transport.util.Request
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+
+class MCCardsAndCarouselTests : BaseMessagingClientTest() {
+
+    @Test
+    fun `when preparePostbackMessage is called then returns StructuredMessage with ButtonResponse`() {
+        val json = Json {
+            ignoreUnknownKeys = true
+        }
+
+        val expectedButtonResponse = QuickReplyTestValues.postbackButtonResponse
+
+        every { mockCustomAttributesStore.getCustomAttributesToSend() } returns emptyMap()
+
+        subject.connect()
+
+        val result = MessageStore(mockLogger).preparePostbackMessage(
+            token = Request.token,
+            buttonResponse = expectedButtonResponse,
+            channel = null
+        )
+
+        val structuredMessage = result.message as StructuredMessage
+        val messageId = structuredMessage.metadata?.get("customMessageId")
+        requireNotNull(messageId) { "customMessageId should not be null" }
+
+        val messageJson = json.encodeToJsonElement(
```

Comment:
We have SerializationTest and RequestSerializationTest classes entirely dedicated for serialization tests. Thats to keep our "functionality" tests clean. 
You can do the serialization testing there. Here you can directly test the values of OnMessageRequest received from preparePostbackMessage() 

Consider to take a look on `when connect() and then sendQuickReply() but no custom attributes`() test in MCQuickRepliesTest. It should pretty similar use case (with just a few tweaks)